src/fastaq_handler.cpp

```diff
@@ -24,16 +24,18 @@ FastaqHandler::FastaqHandler(const std::string& filepath)
         throw "Unable to open " + this->filepath;
     }
     this->inbuf = kseq_init(this->fastaq_file);
+    read_ahead_next_read();
 }
 
-FastaqHandler::~FastaqHandler() { close(); }
+FastaqHandler::~FastaqHandler() {
+    if (!this->is_closed()) {
+        close();
+    }
+}
```

Comment:
This was added to fix a bug in a test case:

https://github.com/mbhall88/pandora/blob/21cd91361ddf2c5310a4aa3633617bde35f46ee5/test/fastaq_handler_test.cpp#L243-L250

This test case closes the `FastaqHandler` on line 248, but then when the object gets destroyed at cleanup, this destructor is called and the `close()` method is called again, which causes an error as the file is already closed. This avoids this errors.

... but as I am writing this now, I just realised that if the user calls `FastaqHandler::close()` more than one time in the same file, it will still cause the error. It might be better to move this check to `FastaqHandler::close()`, so we allow `FastaqHandler::close()` to be called multiple times, along with the destructor, without issuing an error...